### PR TITLE
feat(bigtable): Add support for channel pooling

### PR DESCRIPTION
**Pull request by diptanshumittal**

### CODE REVIEW

File: google-cloud-bigtable/lib/google/cloud/bigtable/service.rb [hunk 1]
```diff
@@ -83,16 +89,14 @@ def tables
...
-            config.endpoint = host if host
-            config.lib_name = "gccl"
-            config.lib_version = Google::Cloud::Bigtable::VERSION
-            config.metadata = { "google-cloud-resource-prefix": "projects/#{@project_id}" }
+          table_key = "#{table_path}_#{app_profile_id}"
+          if @bigtable_clients.get(table_key).nil?
+            bigtable_client = create_bigtable_client table_path, app_profile_id
+            @bigtable_clients.put table_key, bigtable_client
           end
+          @bigtable_clients.get table_key
```
Comment (dazuma) [inline google-cloud-bigtable/lib/google/cloud/bigtable/service.rb:99]: Seems like we could encounter a race condition here if another thread pushes out the table_key after the check on line 95 but before line 99. You'll probably need to synchronize here.

**Review by dazuma — CHANGES_REQUESTED** [review]

File: google-cloud-bigtable/lib/google/cloud/bigtable/service.rb (hunk 1 above)
Comment (diptanshumittal) [reply google-cloud-bigtable/lib/google/cloud/bigtable/service.rb]: Updated to wrap it inside a mutex block.

File: google-cloud-bigtable/lib/google/cloud/bigtable.rb [hunk 2]
```diff
@@ -76,12 +80,14 @@ def self.new project_id: nil,
                    scope: nil,
                    endpoint: nil,
                    endpoint_admin: nil,
-                   timeout: nil
-        project_id    ||= default_project_id
-        scope         ||= configure.scope
-        timeout       ||= configure.timeout
+                   timeout: nil,
+                   channel_selection: configure.channel_selection,
+                   channel_count: configure.channel_count
```
[the remark this answers is not in the document]
Comment (diptanshumittal) [reply google-cloud-bigtable/lib/google/cloud/bigtable.rb]: Following the pattern increased the method complexity. Updated it to use `||=` to implement a default value. For rubocop have the disabled the warnings.

**Review by dazuma — APPROVED** [review]
Thanks!

**Pull request merged**